system-variables: add docs for new variable `tidb_opt_objective` (#14588)
pingcap/docs: system-variables.md

- Default value: `1.0`
- Indicates the net cost of transferring 1 byte of data through the network. This variable is internally used in the [Cost Model](/cost-model.md), and it is **NOT** recommended to modify its value.

### tidb_opt_objective <span class="version-mark">New in v7.4.0</span>

- Scope: SESSION | GLOBAL

> - Persists to cluster: Yes

Reviewer: ```suggestion
- Persists to cluster: Yes
- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): Yes
```

- Type: Enumeration
- Default value: `moderate`
- Possible values: `moderate`, `determinate`

> - This variable controls the objective of the optimizer. `moderate` maintains the default behavior from previous versions, where the optimizer tries to use more information to generate better execution plans. `determinate` mode tends to be more conservative and makes the execution plan more stable.
> - The real-time statistics are the total number of rows and the number of modified rows that are automatically updated based on DML statements. When this variable is set to `moderate` (default), TiDB will generate the execution plan based on real-time statistics. When this variable is set to `determinate`, TiDB won't use real-time statistics for generating the execution plan, which will make execution plans more stable.

Reviewer: ```suggestion
- This variable controls the objective of the optimizer. `moderate` maintains the default behavior in versions prior to TiDB v7.4.0, where the optimizer tries to use more information to generate better execution plans. `determinate` mode tends to be more conservative and makes the execution plan more stable.
- The real-time statistics are the total number of rows and the number of modified rows that are automatically updated based on DML statements. When this variable is set to `moderate` (default), TiDB generates the execution plan based on real-time statistics. When this variable is set to `determinate`, TiDB does not use real-time statistics for generating the execution plan, which will make execution plans more stable.
```

> - For long-term and stable OLTP workload, or the user is affirmative on the existing execution plans, the `determinate` mode is recommended to reduce the possibility of unexpected execution plan changes. The [`LOCK STATS`](/sql-statements/sql-statement-lock-stats.md) can also be used to prevent the statistics from being modified to further stabilize the execution plan.

Reviewer 2: ```suggestion
- For long-term stable OLTP workload, or if the user is affirmative on the existing execution plans, it is recommended to use the `determinate` mode to reduce the possibility of unexpected execution plan changes. Additionally, you can use the [`LOCK STATS`](/sql-statements/sql-statement-lock-stats.md) to prevent the statistics from being modified and further stabilize the execution plan.
```
